src/main/java/dev/lambdaurora/aurorasdeco/mixin/entity/SleepTaskMixin.java

```diff
@@ -33,9 +34,10 @@
 
 @Mixin(SleepTask.class)
 public class SleepTaskMixin {
-	@Inject(method = "shouldKeepRunning", at = @At(value = "RETURN", ordinal = 1), cancellable = true, locals = LocalCapture.CAPTURE_FAILHARD)
-	private void onShouldKeepRunning(ServerWorld world, LivingEntity entity, long time, CallbackInfoReturnable<Boolean> cir,
-	                                 Optional<GlobalPos> homePos, BlockPos pos) {
+	@Inject(method = "shouldKeepRunning", at = @At(value = "RETURN", ordinal = 1), cancellable = true)
+	private void onShouldKeepRunning(ServerWorld world, LivingEntity entity, long time, CallbackInfoReturnable<Boolean> cir) {
+		Optional<GlobalPos> homePos = entity.getBrain().getOptionalMemory(MemoryModuleType.HOME);
```

Comment:
Wait, there's really no way to capture those? :(

Reply:
Output of LocalCapture.PRINT:
```
/*****************************************************************************************************************************/
/*       Target Class : net.minecraft.entity.ai.brain.task.SleepTask                                                         */
/*      Target Method : public boolean shouldKeepRunning(ServerWorld arg0, LivingEntity entity, long arg2)                   */
/*  Target Max LOCALS : 7                                                                                                    */
/* Initial Frame Size : 5                                                                                                    */
/*      Callback Name : onShouldKeepRunning                                                                                  */
/*        Instruction : InjectionNode IRETURN                                                                                */
/*****************************************************************************************************************************/
/*   LOCAL                  TYPE  NAME                                                                                       */
/*   [  0]             SleepTask  this                                                                                       */
/*   [  1]           ServerWorld  arg0                                                                                       */
/*   [  2]          LivingEntity  entity                                                                                     */
/*   [  3]                  long  arg2                                                                                       */
/*   [  4]                 <top>                                                                                             */
/* > [  5]                     -                                                                                             */
/*   [  6]                     -                                                                                             */
/*****************************************************************************************************************************/
/*                                                                                                                           */
/* /**                                                                                                                       */
/*  * Expected callback signature                                                                                            */
/*  * /                                                                                                                      */
/* private void onShouldKeepRunning(ServerWorld arg0, LivingEntity entity, long arg2, CallbackInfoReturnable<Boolean> cir) { */
/*     // Method body                                                                                                        */
/* }                                                                                                                         */
/*                                                                                                                           */
/*****************************************************************************************************************************/
```
I don't have enough mixin knowledge to find or rule out another workaround, but the locals are not available at the current injection point.